### PR TITLE
docs: how it works

### PR DESCRIPTION
**Pull request by luislhl**
## Acceptance Criteria
- Add docs explaining briefly some important topics about how the service works

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -30,3 +30,66 @@ We also have a public Docker image if you don't want to build it yourself:
...
+### A new miner connects
+When a new miner connects, a new instance of the protocol module is created for it and registered in the manager.
+
+It starts 2 periodic tasks:
+
+- Estimator Task: It will adjust the difficulty of the next job to be mined. It will do so with the goal of keeping the time to solve a job equal to the TARGET_JOB_TIME.
+
+This difficulty adjustment is done to allow us to calculate the hashrate of the miner, by checking how many jobs it submitted in a given period and their weight.
+
+It should be noted that because of this mechanism, the difficulty we assign to jobs can be lower than the difficulty of the block template, in the case of block jobs. Especially in the mainnet, where the difficulty of the network is too high.
```
Comment (jansegre) [inline README.md:59]: > ... _can be lower_ than the difficulty of the block template, ...

I think this is a bit of an understatement. I'd say "_will usually be_ lower than the difficulty of the block template" or "_will certainly be_ lower than the difficulty of the block template".

**Review by jansegre — APPROVED** [review]

**Review by pedroferreira1 — APPROVED** [review]

File: README.md [hunk 2]
```diff
@@ -30,3 +30,66 @@ We also have a public Docker image if you don't want to build it yourself:
...
+
+### A new block template is found
+The manager runs a periodic task every 3 seconds to check if there is a new block template.
+
+If a new one is found, it will try to update the current job of all miners, but only if they are currently mining a block.
+If they are mining a tx, it will do nothing.
+
+### A new job is submitted to the API server
+
+When a new job arrives at the API, it will be added to the manager's job queue.
```
Comment (pedroferreira1) [inline README.md:83]: Maybe we could make it explicit that this is only for tx jobs.

Comment (luislhl) [reply README.md:87]: Done

File: README.md (hunk 1 above)
Comment (luislhl) [reply README.md]: Improved

**Review by jansegre — APPROVED** [review]

**Review by pedroferreira1 — APPROVED** [review]

**Pull request merged**